logstash-core/src/main/java/org/logstash/plugins/factory/ExecutionContext.java

```diff
@@ -0,0 +1,77 @@
+package org.logstash.plugins.factory;
+
+import co.elastic.logstash.api.Context;
+import co.elastic.logstash.api.DeadLetterQueueWriter;
+import org.jruby.Ruby;
+import org.jruby.RubyBasicObject;
+import org.jruby.RubyClass;
+import org.jruby.anno.JRubyClass;
+import org.jruby.anno.JRubyMethod;
+import org.jruby.runtime.ThreadContext;
+import org.jruby.runtime.builtin.IRubyObject;
+import org.logstash.RubyUtil;
+import org.logstash.common.AbstractDeadLetterQueueWriterExt;
+import org.logstash.common.DLQWriterAdapter;
+import org.logstash.common.NullDeadLetterQueueWriter;
+import org.logstash.execution.ExecutionContextExt;
+import org.logstash.instrument.metrics.AbstractNamespacedMetricExt;
+import org.logstash.plugins.ContextImpl;
+import org.logstash.plugins.NamespacedMetricImpl;
+import org.logstash.plugins.PluginLookup;
+
+@JRubyClass(name = "ExecutionContextFactory")
+public final class ExecutionContext extends RubyBasicObject {
```

Comment:
maybe naming Ruby/Java joined parts should follow some conventions consistently.
so in Ruby this is a `ExecutionContextFactory` but in Java its missing the `Factory` suffix, why?

Reply:
These was inner classes, I kept the name. To be adherent to convention used it should be named `ExecutionContextExt`

Reply:
Renamed to `ExecutionContextExt`